File: sqfile/src/main/java/com/effective/android/sqfile/SQFile.java
package com.effective.android.sqfile;

import android.content.Context;

import androidx.annotation.WorkerThread;

import com.effective.android.sqfile.bean.InsertParameter;
import com.effective.android.sqfile.bean.SQParameter;
import com.effective.android.sqfile.bean.SQResult;
import com.effective.android.sqfile.interfaces.Callback;
import com.effective.android.sqfile.scope.FileScope;

public class SQFile {

    private static Context sContext;
    FileScope fileScope;
    String dir;
    SQParameter parameter;

    static void init(Context context) {
        sContext = context;
    }

    public static Context getContext() {
        return sContext;
    }

    private SQFile(FileScope fileScope, String dir) {
        this.fileScope = fileScope;
        this.dir = dir;
    }

    public static SQFile create(FileScope fileScope) {
        return new SQFile(fileScope, null);
    }

    public static SQFile create(FileScope fileScope, String appendDir) {
        return new SQFile(fileScope, appendDir);
    }

    public SQFile insert(InsertParameter parameter) {
        this.parameter = parameter;
        return this;
    }


    public void enqueue(Callback responseCallback) {

    }

    @WorkerThread
    public SQResult execute() {
        return null;
    }

    public SQFile delete() {
        return this;
    }

    public SQFile query() {
        return this;
    }
}

File: sqfile/src/main/java/com/effective/android/sqfile/SQFile.kt
package com.effective.android.sqfile

import android.content.Context
import android.os.Handler
import android.os.Looper
import com.effective.android.sqfile.param.*
import com.effective.android.sqfile.scope.FileScope
import java.util.concurrent.LinkedBlockingDeque
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.TimeUnit

object SQFile {

    internal lateinit var handler: Handler
    internal lateinit var threadPoolExecutor: ThreadPoolExecutor
    internal lateinit var context: Context

    /**
     * 初始化context
     */
    internal fun init(context: Context) {
        this.context = context
        this.handler = Handler(Looper.getMainLooper())
        this.threadPoolExecutor = ThreadPoolExecutor(3, 5, 0, TimeUnit.SECONDS, LinkedBlockingDeque<Runnable>(100))
    }

    @JvmStatic
    fun insert(fileScope: FileScope): SQFParameterAdapter<InsertParameter> {
        return SQFParameterAdapter(fileScope)
    }

    @JvmStatic
    fun delete(fileScope: FileScope): SQFParameterAdapter<DeleteParameter> {
        return SQFParameterAdapter(fileScope)
    }

    @JvmStatic
    fun query(fileScope: FileScope): SQFParameterAdapter<QueryParameter> {
        return SQFParameterAdapter(fileScope)
    }

}